### ISSUE TITLE
[New updater] Update from 9.0 to a future version fails.

### ISSUE THREAD

**Issue by jvillafanez** [owncloud/updater#225]
### Steps to reproduce
1. Setup a test server to download the updates from. (*)
2. Modify the version of the core and the "files" app to another one in the future, in the package to be downloaded (**)
3. Run the updater

### Expected behaviour
ownCloud is updated

### Actual behaviour
Update fails

After debugging I've noticed that, during the upgrade, the apps/files/appinfo/info.xml is still the old one (min-version = 9.0 and max-version = 9.0) and the owncloud version is the 9.1. Trying to upgrade is these conditions will disable the "files" app because is not supported, causing the current issue.

Notice that the _oc_upgrade/9.1.0/owncloud/apps/files/appinfo/info.xml contains the right information (min-version = 9.1 and max-version = 9.1).

### Server configuration
**Operating system**: ubuntu 14.04

**Web server:** apache 2.4.7

**Database:** mysql

**PHP version:** 5.5.9

**ownCloud version:** daily-master (OC 9, Feb 8th)

**Updated from an older ownCloud or fresh install:** fresh install

### Updater output

```
ownCloud updater 1.0 - CLI based ownCloud server upgrades
Checking system health.
 - file permissions are ok.
Current version is 9.0.0.10
9.1.0 pre alpha is found online
What would you do next?
  [0] download
  [1] upgrade
  [2] abort
 > 1
Downloaded 100% (27257922 of 27257922)
Created checkpoint 9.0.0.10-56b88f03bd772
 simulated (optionally, can be done online in advance)
real [danger, might take long]
Extracting source into ./_oc_upgrade/9.1.0


                                                                               
  [Symfony\Component\Process\Exception\ProcessFailedException]                 
  The command "php ./occ upgrade" failed.                                      
  Exit Code: 5(Unknown error)                                                  
  Output:                                                                      
  ================                                                             
  ownCloud or one of the apps require upgrade - only a limited number of comm  
  ands are available                                                           
  You may use your browser or the occ upgrade command to do the upgrade        
  Set log level to debug                                                       
  Turned on maintenance mode                                                   
  Checking whether the database schema can be updated (this can take a long t  
  ime depending on the database size)                                          
  Checked database schema update                                               
  Checking updates of apps                                                     
  Checked database schema update for apps                                      
  Updating database schema                                                     
  Updated database                                                             
  Exception: files can't be disabled.                                          
  Update failed                                                                
  Maintenance mode is kept active                                              
  Reset log level                                                              
  Error Output:                                                                
  ================                                                             
                                                                               


upgrade:executeCoreUpgradeScripts

```

### Notes

(*) The setup is similar to the following:

1. Create a folder in your own ownCloud server, in the base directory of the server (let's call it update-server)
2. Create an index.php file inside with the following content (it's enough for testing):

    ```
<?xml version="1.0" ?>
<owncloud>
<version>9.1.0</version>
<versionstring>9.1.0 pre alpha</versionstring>
<url>http://localhost/update-server/owncloud-fake.tar.bz2</url>
<web>http://www.google.com</web>
</owncloud>
    ```

3. Modify https://github.com/owncloud/updater/blob/master/src/Utils/Fetcher.php#L29 to point to your server (http://localhost/update-server should be enough)

4. Download the daily-master package and modify the version number before leaving it in the localhost/update-server folder (open the tarball, modify the file and compress it again)

    ```
<?php 
$OC_Version = array(9,1,0,10);
$OC_VersionString = '9.1 pre alpha';
$OC_Channel = 'daily';
$OC_VersionCanBeUpgradedFrom = array(9,0);
$OC_Build = '2016-02-08T04:40:38+00:00 feb0522cd059c44b77b5afe1382e5f62bbeb59aa';
    ```

(**) The modification in the "files" app is just to update the supported ownCloud version. Go to the apps/files/appinfo/info.xml file (in the package that the updater will download) and modify the dependency:

```
        <dependencies>
                <owncloud min-version="9.1" max-version="9.1" />
        </dependencies>
```

@VicDeo Looks like this is a sev1

**Comment by VicDeo** [comment]
@jvillafanez may I ask you to post `/path/to/owncloud/updater/update.log` content?

**Comment by rperezb** [comment]
@VicDeo @jvillafanez is on holidays the rest of the week, I will send to you access to the test servers

**Comment by cmonteroluque** [comment]
@VicDeo update please

**Comment by VicDeo** [comment]
@jvillafanez As I see  your `/opt/owncloud/fake-server/owncloud-fake.tar.bz2` package has the following `version.php` file:
```
<?php 
$OC_Version = array(30,0,0,10);
$OC_VersionString = '30.0 pre alpha';
```

and the files app is indeed incompatible with **v30.0** because it is greater than requirements for files app:
```
        <dependencies>
                <owncloud min-version="9.1" max-version="9.1" />
        </dependencies>
```

and it is indeed not possible to disable files app. 

Thus the package is not consistent. (Not only for updater app, try to update manually and you will have the same results)

**Comment by DeepDiver1975** [comment]
> and the files app is indeed incompatible with v30.0 because it is greater than requirements for files app:

this is an invalid test scenario - a package fir version 30.0 has to contain proper code for version 30.0 as well. In this case the contained code was for 9.1.

This can never work

**Comment by jvillafanez** [comment]
The other server (instance id 1652, container name upgrade_test2) has the right versioning, but still not working

```
<?php 
$OC_Version = array(9,1,0,10);
$OC_VersionString = '9.1 pre alpha';
```

```
        <dependencies>
                <owncloud min-version="9.1" max-version="9.1" />
        </dependencies>
```

**Comment by VicDeo** [comment]
@jvillafanez please post `version.php` from your tarball, not OC instance

**Comment by jvillafanez** [comment]
tarball's version.php:

```
<?php 
$OC_Version = array(9,1,0,10);
$OC_VersionString = '9.1 pre alpha';
$OC_Channel = 'daily';
$OC_VersionCanBeUpgradedFrom = array(9,0);
$OC_Build = '2016-02-08T04:40:38+00:00 feb0522cd059c44b77b5afe1382e5f62bbeb59aa';
```

instance's version.php

```
<?php 
$OC_Version = array(9,0,0,10);
$OC_VersionString = '9.0 pre alpha';
$OC_Channel = 'daily';
$OC_VersionCanBeUpgradedFrom = array(8,2);
$OC_Build = '2016-02-08T04:40:38+00:00 feb0522cd059c44b77b5afe1382e5f62bbeb59aa';
```

**Comment by VicDeo** [comment]
@jvillafanez why you are not using beta for tests btw?

**Comment by jvillafanez** [comment]
I tested with the daily package of that day (as you can see in the OC_Build variable :smile: ), so no beta version at that time.

**Comment by VicDeo** [comment]
Well I can't reproduce it with my
```
<?php 
$OC_Version = array(9,0,0,11);
$OC_VersionString = '9.0 beta 1';
$OC_Channel = 'daily';
$OC_VersionCanBeUpgradedFrom = array(8,2);
$OC_Build = '2016-02-16T04:41:41+00:00 907430a80812bbf5378a4c5d10de6ab7ddf6f8e4';
```

**Comment by jvillafanez** [comment]
I think you'll need a custom tarball, with a different version. If you use the official channel it will work because the new ownCloud version (9.0.0.11 in your case) is within the range of the "old" files app version (9.0 for min and max versions), so the files app won't be disabled.

If the new ownCloud version is outside of that range (let's say the yet-to-be-born ownCloud 9.1) the upgrade will fail even if the "new" files app (the one that will be bundle in ownCloud 9.1) version is 9.1 because the "old" files app is 9.0.

When the upgrade starts, the upgrade checks the version of the current installed files app (9.0) even if the version.php is 9.1, because the new files app (with the 9.1 version) isn't yet in place. The upgrade process will disable the files app because it's not supported, causing the current issue.

**Comment by VicDeo** [comment]
@jvillafanez 
>When the upgrade starts, the upgrade checks the version of the current installed files app (9.0) even if the version.php is 9.1, because the new files app (with the 9.1 version) isn't yet in place. The upgrade process will disable the files app because it's not supported, causing the current issue.

makes sense, will check

**Comment by VicDeo** [comment]
@DeepDiver1975 ok, this is seems valid. 
When upgrading 
to 9.1 with files app `require-min`=9.1  
from 9.0 with files app `require-max`=9.0 
new app versions are added after core upgrade step and core is desperately trying to disable all incompatible apps first and stucks on files app that is not possible to disable.

**Comment by VicDeo** [comment]
@jvillafanez addressed in #256 

 `cat apps/files/appinfo/info.xml`
```
...snip
        <dependencies>
                <owncloud min-version="9.1" max-version="9.1" />
        </dependencies>
...snip
```
Before
![before](https://cloud.githubusercontent.com/assets/991300/13354530/ab0cafaa-dcac-11e5-902b-e05cff4a8b65.png)

core upgrade step
![progress](https://cloud.githubusercontent.com/assets/991300/13354604/178af9a2-dcad-11e5-9e4f-84c56bb85d4b.png)

Admin after upgrade
![after](https://cloud.githubusercontent.com/assets/991300/13354536/b1d55076-dcac-11e5-9cf8-668616f6a1e3.png)
